Source/WebCore/platform/graphics/cg/GradientRendererCG.cpp

```diff
@@ -485,11 +493,20 @@ GradientRendererCG::Strategy GradientRendererCG::makeGradient(ColorInterpolation
         // extended sRGB for all gradients.
         if (hasOnlyBoundedSRGBColorStops(stops)) {
             for (const auto& stop : stops) {
-                auto [r, g, b, a] = stop.color.toColorTypeLossy<SRGBA<float>>().resolved();
-                colorComponents.appendList({ r, g, b, a });
+                if (destinationColorSpace) {
+                    auto [r, g, b, a] = stop.color.toResolvedColorComponentsInColorSpace(*destinationColorSpace);
+                    colorComponents.appendList({ r, g, b, a });
+                } else {
+                    auto [r, g, b, a] = stop.color.toColorTypeLossy<SRGBA<float>>().resolved();
+                    colorComponents.appendList({ r, g, b, a });
+                }
 
                 locations.append(stop.offset);
             }
+
+            if (destinationColorSpace)
+                return destinationColorSpace->platformColorSpace();
+
```

Comment:
How does this all work with the OutputSpaceColorType used in GradientRendererCG::makeShading? I would have expected that needed to change as well if we weren't always going to use sRGB or extendedSRGB anymore.

Reply:
Using the destination colorspace is optional; it's just an optimization, so the less common shading path doesn't need to do it.